### PR TITLE
INTMDB-416 database user

### PR DESCRIPTION
**Pull request by ParthasarathyVaradhan**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ✅ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

## Further comments
Adding Example AWS template for database users

### CODE REVIEW

File: examples/database-user/user.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Description": "Returns all database users that belong to the specified project. To use this resource, the requesting API Key must have the Project Read Only role. This resource doesn't require the API Key to have an Access List.",
+  "Parameters": {
+    "ProjectId": {
```
Comment (Zuhairahmed) [inline examples/database-user/user.json:5]:
```suggestion
    "ProjectID": {
```

File: examples/database-user/user.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      "Type": "String",
+      "Description": "",
+      "ConstraintDescription": ""
+    },
+    "PublicKey": {
+      "Type": "String",
+      "Description": "",
+      "ConstraintDescription": ""
+    },
+    "Privatekey": {
```
Comment (Zuhairahmed) [inline examples/database-user/user.json:15]:
```suggestion
    "PrivateKey": {
```

**Review by Zuhairahmed — APPROVED** [review]
Few minor comments below, not blocking. Also nice work on adding IAM / LDAP / X.509 . SCRAM examples! very helpful for users

**Review by govardhanpagidi — APPROVED** [review]
LGTM!

**Review by matt-condon — APPROVED** [review]

**Pull request merged**